snapcraft/plugins/v2/_ros.py

```diff
@@ -0,0 +1,180 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import abc
+import os
+import pathlib
+import subprocess
+import sys
+from typing import Dict, List, Set
+
+import click
+from catkin_pkg import packages as catkin_packages
+
+from snapcraft.internal.repo import Repo
+from snapcraft.plugins.v1._ros.rosdep import _parse_rosdep_resolve_dependencies
+from snapcraft.plugins.v2 import PluginV2
+
+
+class RosPlugin(PluginV2):
```

Comment:
if this is not intended by base users and only meant to be inherited from, how about making it an ABC?

Reply:
Oh it already is. `PluginV2` is an ABC, and this class even defines more abstract methods.

Reply:
Nice! Missed that part